code/__HELPERS/string_assoc_list.dm

```diff
@@ -0,0 +1,17 @@
+GLOBAL_LIST_EMPTY(string_assoc_lists)
+
+/**
+ * Caches associative lists with non-numeric stringify-able index keys and stringify-able values (text/typepath -> text/path/number).
+ */
+/datum/proc/string_assoc_list(list/values)
+	var/list/string_id = list()
+	for(var/val in values)
+		string_id += "[val]_[values[val]]"
+	string_id = string_id.Join("-")
+
+	. = GLOB.string_assoc_lists[string_id]
+
+	if(.)
+		return
+
+	return GLOB.string_assoc_lists[string_id] = values
```

Comment:
```dm
/turf/closed/wall/Initialize(mapload)
	. = ..()
	if(is_station_level(z))
		GLOB.station_turfs += src
	if(smoothing_flags & SMOOTH_DIAGONAL_CORNERS && fixed_underlay) //Set underlays for the diagonal walls.
		var/mutable_appearance/underlay_appearance = mutable_appearance(layer = TURF_LAYER, plane = FLOOR_PLANE)
		if(fixed_underlay["space"])
			underlay_appearance.icon = 'icons/turf/space.dmi'
			underlay_appearance.icon_state = SPACE_ICON_STATE
			underlay_appearance.plane = PLANE_SPACE
		else
			underlay_appearance.icon = fixed_underlay["icon"]
			underlay_appearance.icon_state = fixed_underlay["icon_state"]
		fixed_underlay = string_assoc_list(fixed_underlay)
		underlays += underlay_appearance
```

Could you justify the existance of this proc? It seems completely unnecessary, it doesn't even look like it does anything apart from use processing time.

(Also why is this a datum proc when it doesnt use any datum values)